lib/event_listeners.js

```diff
@@ -469,14 +469,55 @@ AWS.EventListeners = {
       var req = resp.request;
       var logger = req.service.config.logger;
       if (!logger) return;
+      var Shape = require('./model/shape');
+      function filterSensitiveLog(inputShape, shape) {
+        switch (inputShape.type) {
```

Comment:
I think this function could be more compact if the switch had a default case (to match scalar types) and it simply recurred for each structure, list, and map member